### ISSUE TITLE
Pipe redirection broken on Python2.7 since 4.3.13

### ISSUE THREAD

**Issue by ionelmc** [timothycrosley/isort#948]
All versions since 4.3.13 (including) produce this problem:
```bash
$ echo "import stuff" | isort -
Traceback (most recent call last):
  File "/usr/local/bin/isort", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/dist-packages/isort/main.py", line 339, in main
    SortImports(file_=file_, write_to_stdout=True, **arguments)
  File "/usr/local/lib/python2.7/dist-packages/isort/isort.py", line 140, in __init__
    file_.seek(0)
IOError: [Errno 29] Illegal seek
```

It would look like https://github.com/timothycrosley/isort/issues/895 is the same bug but then 3.3.15 still produces this problem.

I'll be using 4.3.12 meanwhile ...

**Comment by timothycrosley** [comment]
Hi @ionelmc,

Thanks for reporting this issue:

After some investigation, I've confirmed it was fixed for Python3+, but remains broken on Python2:

Python3:
```
timothy@timothy-Desktop:~$ sudo pip3 install -U isort
Requirement already up-to-date: isort in /usr/local/lib/python3.7/dist-packages (4.3.19)
timothy@timothy-Desktop:~$ echo "import stuff" | isort -
import stuff
```

Python2.7:
```
(test_isort_python2) timothy@timothy-Desktop:~$ pip install isort -U
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Collecting isort
  Downloading https://files.pythonhosted.org/packages/ae/ae/5ef4b57e15489754b73dc908b656b02ab0e6d37b190ac78dd498be8b577d/isort-4.3.19-py2.py3-none-any.whl (42kB)
     |████████████████████████████████| 51kB 3.7MB/s 

(test_isort_python2) timothy@timothy-Desktop:~$ echo "import stuff" | isort -
Traceback (most recent call last):
  File "/home/timothy/test_isort_python2/bin/isort", line 10, in <module>
    sys.exit(main())
  File "/home/timothy/test_isort_python2/local/lib/python2.7/site-packages/isort/main.py", line 343, in main
    SortImports(file_=file_, write_to_stdout=True, **arguments)
  File "/home/timothy/test_isort_python2/local/lib/python2.7/site-packages/isort/isort.py", line 140, in __init__
    file_.seek(0)
IOError: [Errno 29] Illegal seek
```
I'll work on a hotfix release to fix the Python2 behavior as well shortly.

Thanks!

~TImothy

**Comment by timothycrosley** [comment]
This has been fixed in isort 4.3.20:

```(test_isort_python2) timothy@timothy-Desktop:~/bin$ pip install isort -U
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Collecting isort
  Downloading https://files.pythonhosted.org/packages/1c/d9/bf5848b376e441ff358a14b954476423eeeb8c9b78c10074b7f53ce2918d/isort-4.3.20-py2.py3-none-any.whl (42kB)
     |████████████████████████████████| 51kB 5.6MB/s 
Requirement already satisfied, skipping upgrade: backports.functools-lru-cache; python_version < "3.2" in /home/timothy/test_isort_python2/lib/python2.7/site-packages (from isort) (1.5)
Requirement already satisfied, skipping upgrade: futures; python_version < "3.2" in /home/timothy/test_isort_python2/lib/python2.7/site-packages (from isort) (3.2.0)
Installing collected packages: isort
  Found existing installation: isort 4.3.19
    Uninstalling isort-4.3.19:
      Successfully uninstalled isort-4.3.19
Successfully installed isort-4.3.20

(test_isort_python2) timothy@timothy-Desktop:~/bin$ echo "import      os" | isort -
import os
```

Thanks!

~Timothy
